util-core/src/main/scala/com/twitter/util/Function.scala

```diff
@@ -47,6 +47,38 @@ object Function {
    * Creates `() => A` function from given `Callable`.
    */
   def ofCallable[A](c: Callable[A]): () => A = () => c.call()
+
+  /**
+   * Creates a T => R from a JavaFunction. Used for easier interop
+   * between Java 8 and Twitter Util libraries.
+   */
+  def func[T, R](f: JavaFunction[T, R]) = new Function[T, R] {
+    override def apply(value: T): R = f(value)
+  }
+
+  /**
+   * Creates a T => Unit from a JavaConsumer.
+   * Useful for e.g. future.onSuccess
+   */
+  def cons[T](f: JavaConsumer[T]) = new Function[T, Unit] {
+    override def apply(value: T): Unit = f(value)
+  }
+
+  /**
+   * like func, but deals with checked exceptions as well
+   */
+  def exfunc[T, R](f: ExceptionalJavaFunction[T, R]) = new ExceptionalFunction[T, R] {
```

Comment:
My argument for the short / DSL-y version is that for java users interoperating with scala, they'll probably use these methods all the time.  DSLs work well for those kinds of users.  On the other hand, for folks who only rarely need to use this lib, it's simpler for the name to explain what it's for.  I don't think alias-ing is such a sin.